lib/templateFactory.js

```diff
@@ -6,6 +6,9 @@ const compareVersions = require('compare-versions');
 const schema = require('screwdriver-data-schema');
 let instance;
 
+const TEMPLATE_NAME_REGEX = schema.config.regex.FULL_TEMPLATE_NAME;
+const TEMPLATE_NAME_MATCH = 1;
```

Comment:
I feel like it might make more sense to have them in the schema since that's where the regexes themselves are defined. In this file they seem like magic numbers.

To avoid having to import five separate constants for all the different groups they could possibly also be wrapped in an object called `FULL_TEMPLATE_NAME_GROUPS` or something like that.